lib/abstractions/create.js

```
@@ -48,8 +48,32 @@ module.exports = options => {
       return;
     }
 
-    var savedRecord = await pendingRecord.save(),
-        body = {};
+    try {
+      var savedRecord = await pendingRecord.save();
+    } catch ( err ) {
+      let error = err.message;
+
+      if (err.code === 11000){
+        let path = /index:\s([^\d_]*).*"(\w+)/i.exec(error)[1];
+        let val = /index:\s([^\d_]*).*"(\w+)/i.exec(error)[2];
+        let errors = [];
+
+        errors.push({
+          status: 400,
+          title: 'duplicate Key Error',
+          detail: 'duplicate URL Prefix',
+          path,
+          val
+        });
+
+        ctx.status = 400;
+        ctx.body = { errors };
+        return;
+      }
+      ctx.status = 500;
```

You need to throw the error here.

---

like this?     
 ```
ctx.status = 500;
 throw err;

```